pftools/python/parflow/tools/builders.py

```diff
@@ -534,5 +695,389 @@ def get_table(self, props_in_header=True, column_separator='  '):
+            self.run.Geom[domain_geom_name].Lower.Y = bounds[2]
+            self.run.Geom[domain_geom_name].Upper.Y = bounds[3]
+            self.run.Geom[domain_geom_name].Lower.Z = bounds[4]
+            self.run.Geom[domain_geom_name].Upper.Z = bounds[5]
+
+        if patches:
+            self.run.Geom[domain_geom_name].Patches = patches
+
+        return self
+
+    def slopes_mannings(self, domain_geom_name, slope_x=None,
+                        slope_y=None, mannings=None):
+        """Setting slopes and mannings coefficients as constant value
+        or from an external file
+        """
+        if slope_x is not None:
+            self.run.TopoSlopesX.GeomNames = domain_geom_name
+            if isinstance(slope_x, str):
+                self.__file_check(slope_x, 'TopoSlopesX')
+            else:
+                self.run.TopoSlopesX.Type = 'Constant'
+                self.run.TopoSlopesX.Geom[domain_geom_name].Value = slope_x
+        if slope_y is not None:
+            self.run.TopoSlopesY.GeomNames = domain_geom_name
+            if isinstance(slope_y, str):
+                self.__file_check(slope_y, 'TopoSlopesY')
+            else:
+                self.run.TopoSlopesY.Type = 'Constant'
+                self.run.TopoSlopesY.Geom[domain_geom_name].Value = slope_y
+        if mannings is not None:
+            self.run.Mannings.GeomNames = domain_geom_name
+            if isinstance(mannings, str):
+                self.__file_check(mannings, 'Mannings')
+            else:
+                self.run.Mannings.Type = 'Constant'
+                self.run.Mannings.Geom[domain_geom_name].Value = mannings
+
+        return self
+
+    def zero_flux(self, patches, cycle_name, interval_name):
+        """Setting zero-flux boundary condition for patch or patches
+        """
+        if not self.run.BCPressure.PatchNames:
+            self.run.BCPressure.PatchNames = []
+
+        for patch in patches.split():
+            self.run.BCPressure._details_['PatchNames']['history'] = []
+            self.run.BCPressure.PatchNames += [patch]
+            self.run.Patch[patch].BCPressure.Type = 'FluxConst'
+            self.run.Patch[patch].BCPressure.Cycle = cycle_name
+            self.run.Patch[patch].BCPressure[interval_name].Value = 0.0
+
+        return self
+
+    def ic_pressure(self, domain_geom_name, patch, pressure):
+        """Setting initial condition pressure from file or to constant value
+        """
+        self.run.ICPressure.GeomNames = domain_geom_name
+        self.run.Geom[domain_geom_name].ICPressure.RefPatch = patch
+
+        if isinstance(pressure, str) and get_file_extension(pressure) == 'pfb':
+            self.run.ICPressure.Type = 'PFBFile'
+            self.run.Geom.domain.ICPressure.FileName = pressure
+        elif isinstance(pressure, float) or isinstance(pressure, int):
+            self.run.ICPressure.Type = 'HydroStaticPatch'
+            self.run.Geom.domain.ICPressure.Value = pressure
```

Comment:
Should there be an `else` block that produces an error, in case it doesn't go through either of the other two blocks?